public/dorequest.php

```diff
@@ -805,6 +787,10 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
             break;
         }
 
+        if (VirtualGameIdService::isVirtualGameId($gameID)) {
+            [$gameID, $compatibility] = VirtualGameIdService::decodeVirtualGameId($gameID);
+        }
```

Comment:
This remaps the virtual ID back to the regular ID for the `uploadachievement` API. The same change has been applied to `codenotes2` and `submitcodenote` in the new classes.